packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -451,36 +425,118 @@ export const Tabs = (props: TabsProps) => {
     ownerDoc.addEventListener('scroll', determineScrollingPlacement);
   }, [overflowOptions?.useOverflow, determineScrollingPlacement]);
 
-  const onTabClick = React.useCallback(
-    (index: number) => {
-      if (onTabSelected) {
-        onTabSelected(index);
-      }
-      setCurrentActiveIndex(index);
-    },
-    [onTabSelected],
+  return (
+    <Box
+      className={cx(
+        'iui-tabs',
+        `iui-${type}`,
+        {
+          'iui-green': color === 'green',
+          'iui-animated': type !== 'default' && isClient,
+          'iui-not-animated': type !== 'default' && !isClient,
+          'iui-large': hasSublabel,
+        },
+        className,
+      )}
+      data-iui-overflow={overflowOptions?.useOverflow}
+      data-iui-scroll-placement={scrollingPlacement}
+      role='tablist'
+      ref={refs}
+      {...rest}
+    >
+      {items.map((item, index) => {
+        return (
+          <TabsTabListContext.Provider
```

Comment:
This context provider passes props that are consumed by both `Tab` and `TabDescription`. I figured the context was named according to where it's be provided instead of where it's being consumed.

Does this mean I should wrap the items in 2 sets of Contexts?

Reply:
> I figured the context was named according to where it's be provided instead of where it's being consumed.

the name should be based on where it is used, yes, but in this case you are using it inside a loop, so it's for individual tabs, not the whole tabslist.

---

but i still think the [comment from earlier](https://github.com/iTwin/iTwinUI/pull/1401#discussion_r1275298092) is not completely addressed - most things inside this context should *not* be necessary, except for `hasSublabel`/`setHasSublabel`. and even in that case, you only need one context, not a loop of several contexts.

Reply:
After the addition of the `value` prop in the `Tab` and `Panel` subcomponents, there is no longer a need to loop the `TabListContext` for each individual tab.

Outside of `hasSublabel` / `setHasSublabel`, here are the remaining props I am still passing through context:

- `onTabClick`: A function that clicks a tab given its index. Since tab focusing is done through `Tab`, I needed a way to click on the next available tab given its index when `focusActivationMode` is set to `auto`.
- `setFocusedValue`: Called in the `onClick` and `focusTab` functions to set the new focused value.
- `focusActivationMode`: Prop in `Tabs.TabList` used to check if the tab is clicked when focusing (`'auto'`) or when pressing `Enter` or `Spacebar` (`'manual'`).